Migrate Jest Puppeteer E2E tests to Playwright

gigitux opened the pull request:
> [!WARNING]  
> This PR is blocked by #46125 and #46644

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This PR migrates (and improves) Jest Puppetter E2E tests to Playwright for these blocks:

- Product Categories
- Product Category
- Product new
- Product On Sale
- Product results Count
- Product Tag
- Product Top Rated
- Products By Attribute
- Single Product Details
- Store Notices

Part of #46579.

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure that the CI jobs about E2E tests are green.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

Migrate Jest Puppeteer E2E tests to Playwright.

</details>

WunderBart committed:
try db reset in page teardown

WunderBart committed:
move reset to setup step

WunderBart committed:
Use wp db cli

WunderBart committed:
Fix global setup
That part is overriding the logged in user state and wipes out the nonce and rootUrl fields.

WunderBart committed:
Try importing instead from a generated dump

WunderBart committed:
Revert "Try importing instead from a generated dump"

This reverts commit 987dc471c9dab49dd14aa8a96be43418454133a1.

WunderBart committed:
Revert "Revert "Try importing instead from a generated dump""

This reverts commit c8d008cb2040c03c4cebc725734856fc5a472017.

WunderBart committed:
Don't bypass visitSiteEditor so that the Welcome Guide is closed

gigitux committed:
use createNewPost

WunderBart committed:
Revert "Revert "Revert "Try importing instead from a generated dump"""

This reverts commit 26842735822e95252fbe1cc63b026db4c1949f16.

WunderBart committed:
Merge remote-tracking branch 'origin' into try/e2e-db-reset

WunderBart committed:
Merge remote-tracking branch 'origin' into try/e2e-db-reset

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce into try/e2e-db-reset

gigitux committed:
[Blocks]: Fix E2E tests (#46344)

* Load local pickup enabled setting as bool not string

* Add changelog

* Remove unused import

* Update plugins/woocommerce-blocks/assets/js/extensions/shipping-methods/pickup-location/utils.ts

Co-authored-by: Niels Lange <info@nielslange.de>

* Use strict equality test

* try now

* try now

* add a new sql file

* remove default.sql

* fix welcome guide tour

* fix E2E tests

* Fix the template revert tests
...where the template is unreachable due to pagination.

* Add changelog entry

* improve logic to closeWelcomeGuideModal

* fix cart checkout tests

* improve flakiness

* improve flakiness

* fix flaky test

* fix company field

* fix E2E test

* fix E2E tests

---------

Co-authored-by: Thomas Roberts <thomas.roberts@automattic.com>
Co-authored-by: Thomas Roberts <5656702+opr@users.noreply.github.com>
Co-authored-by: Niels Lange <info@nielslange.de>
Co-authored-by: Bart Kalisz <bartlomiej.kalisz@gmail.com>

WunderBart committed:
Replace beforeAll w/ beforeEach + remove all after* hooks

WunderBart committed:
Fix broken hooks

WunderBart committed:
Activate plugins via requestUtils API

gigitux committed:
[Blocks - E2E]: Add `playwright/no-hooks` ESlint rule (#46432)

add ESLint configuration

WunderBart committed:
Clean up console logs

WunderBart committed:
Remove obsolete language setup steps

WunderBart committed:
Remove more unnecessary setup steps

WunderBart committed:
Remove even more obsolete setup steps

WunderBart committed:
tmp: add the LYS fix

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce into try/e2e-db-reset

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce into try/e2e-db-reset

WunderBart committed:
Try stabilizing the company field test

WunderBart committed:
Use canvas param instead of manually entering edit mode

WunderBart committed:
Remove double site editor redirect

WunderBart committed:
Revert "Use canvas param instead of manually entering edit mode"

This reverts commit 5e6cc1715490ed333b2e363e1cdcefe4e8696683.

WunderBart committed:
Revert "Remove double site editor redirect"

This reverts commit 69a57a82a86ab0fe39002bc59838e6a63d118298.

WunderBart committed:
Fix flaky products sorting test

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce into try/e2e-db-reset

gigitux committed:
Blocks: Fix ESLint errors (#46595)

fix eslint error

gigitux committed:
Fix ESLint errors (#46626)

* Fix ESLint errors

* Add changefile(s) from automation for the following project(s): woocommerce-blocks

---------

Co-authored-by: github-actions <github-actions@github.com>

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce into fix/complete-migration

gigitux committed:
Blocks: Migrate multiple tests to Playwright

gigitux committed:
Merge branches 'try/e2e-db-reset' and 'trunk' of github.com:woocommerce/woocommerce into try/e2e-db-reset

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce into fix/migration-to-pw-pt-1

gigitux committed:
Merge branch 'try/e2e-db-reset' of github.com:woocommerce/woocommerce into fix/migration-to-pw-pt-1

gigitux committed:
fix path

gigitux committed:
migrate other tests

gigitux committed:
add product best sellers E2E tests

Commit:
Add changefile(s) from automation for the following project(s): woocommerce-blocks

gigitux committed:
migrate jest-puppetter E2E tests to Playwright

Commit:
Add changefile(s) from automation for the following project(s): woocommerce-blocks

Commit:
Add changefile(s) from automation for the following project(s): woocommerce-blocks

Commit:
Add changefile(s) from automation for the following project(s): woocommerce-blocks

WunderBart committed:
Merge remote-tracking branch 'origin' into try/e2e-db-reset

WunderBart committed:
Remove obsolete waitForLoadState call

WunderBart committed:
Merge remote-tracking branch 'origin' into try/e2e-db-reset

Aljullu commented on plugins/woocommerce-blocks/tests/e2e/tests/product-results-count/product-results-count.block_theme.spec.ts:
```diff
@@ -0,0 +1,36 @@
+		await editorUtils.enterEditMode();
+		await expect(
+			editorUtils.insertBlockUsingGlobalInserter( blockData.name )
+		).resolves.not.toThrow();
```
Besides checking that it doesn't throw, what do you think about also checking it renders correctly by checking the rendered text, for example?

gigitux replied:
Good point! Fixed [490b49c](https://github.com/woocommerce/woocommerce/pull/46671/commits/490b49c02ee991b696dd18331921f61de3f05508)

Aljullu commented on plugins/woocommerce-blocks/tests/e2e/tests/single-product-details/single-product-details.block_theme.spec.ts:
```diff
@@ -0,0 +1,39 @@
+		await editorUtils.enterEditMode();
+		await expect(
+			editorUtils.insertBlockUsingGlobalInserter( blockData.name )
+		).resolves.not.toThrow();
```
Same here.

gigitux replied:
Good point! Fixed [490b49c](https://github.com/woocommerce/woocommerce/pull/46671/commits/490b49c02ee991b696dd18331921f61de3f05508)

Aljullu commented on plugins/woocommerce-blocks/tests/e2e/tests/store-notices/store-notices.block_theme.spec.ts:
Should the equivalent `/tests/e2e-jest/specs/backend/store-notices.test.js` be removed?

gigitux replied:
Good catch! I removed it with [e504b16](https://github.com/woocommerce/woocommerce/pull/46671/commits/e504b16160875e337816d7ad848c736ce1877305)

Aljullu commented on plugins/woocommerce-blocks/tests/e2e/tests/product-category/product-category.block_theme.spec.ts:
```diff
@@ -0,0 +1,30 @@
+		await admin.createNewPost();
+		await editor.insertBlock( { name: blockData.slug } );
+		const blockLocator = await editorUtils.getBlockByName( blockData.slug );
+		await expect( blockLocator.getByRole( 'listitem' ) ).toHaveCount( 9 );
```
Shouldn't we select a category in this test similar to what's done in `product-tag`? Otherwise only the "Select category" screen is shown.

gigitux replied:
Yep! I addressed with [cceae5e](https://github.com/woocommerce/woocommerce/pull/46671/commits/cceae5e40ed8971782d3d6edd5888934a33a7fe0)

Aljullu commented on plugins/woocommerce-blocks/tests/e2e/tests/product-on-sale/product-on-sale.block_theme.spec.ts:
```diff
@@ -0,0 +1,30 @@
+		await admin.createNewPost();
+		await editor.insertBlock( { name: blockData.slug } );
+		const blockLocator = await editorUtils.getBlockByName( blockData.slug );
+		await expect( blockLocator.getByRole( 'listitem' ) ).toHaveCount( 9 );
```
This test seems to be failing. I think there are only 6 on-sale products instead of 9.

gigitux replied:
Fixed with: [d1cee4f](https://github.com/woocommerce/woocommerce/pull/46671/commits/d1cee4f4e8c181a9de792dced8e5d72c40eb5fbc)

Aljullu commented on plugins/woocommerce-blocks/tests/e2e/tests/single-product-details/single-product-details.block_theme.spec.ts:
```diff
@@ -0,0 +1,39 @@
+		admin,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'woocommerce/woocommerce//archive-product',
```
This block is only available in the Single Product template.

gigitux replied:
Fixed with [490b49c](https://github.com/woocommerce/woocommerce/pull/46671/commits/490b49c02ee991b696dd18331921f61de3f05508)

Aljullu commented on plugins/woocommerce-blocks/tests/e2e/tests/store-notices/store-notices.block_theme.spec.ts:
```diff
@@ -0,0 +1,39 @@
+};
+
+test.describe( `${ blockData.slug } Block`, () => {
+	test( "block can't be inserted in Post Editor", async ( {
```
I'm not sure if that's a bug or intended, but the Store Notices block seems to be available in the post editor. :thinking:

gigitux replied:
Mmm, interesting! I can confirm that I'm able to add the Store Notice block in the Post Editor. I refactored it with [94210f9](https://github.com/woocommerce/woocommerce/pull/46671/commits/94210f976717daab873bb1abf8b93580e7811439): there was something wrong with the old test.

Aljullu reviewed (commented):
Thanks for working on this, @gigitux! Very exciting to see these tests being migrated to Playwright!

I left some small comments, some of them are related to some tests which are failing on my end. Let me know if you can't reproduce and I will investigate them in more detail!

gigitux committed:
Merge branch 'try/e2e-db-reset' of github.com:woocommerce/woocommerce into fix/migration-to-pw-pt-1

gigitux committed:
improve E2E tests

gigitux committed:
Merge branch 'fix/migration-to-pw-pt-1' of github.com:woocommerce/woocommerce into fix/migration-to-pw-pt-1

gigitux committed:
fix E2E tests

gigitux committed:
Merge branch 'fix/migration-to-pw-pt-1' of github.com:woocommerce/woocommerce into fix/migration-to-pw-pt2

gigitux committed:
improve E2E tests

gigitux committed:
remove old test

gigitux committed:
remove not necessary test

gigitux committed:
Merge branch 'fix/migration-to-pw-pt2' of github.com:woocommerce/woocommerce into fix/migration-to-pw-pt2

gigitux committed:
fix number product

gigitux committed:
fix E2E test

gigitux committed:
fix number of products

gigitux committed:
Merge branch 'fix/migration-to-pw-pt-1' of github.com:woocommerce/woocommerce into fix/migration-to-pw-pt2

gigitux committed:
fix product count

gigitux committed:
refactor store notice E2E tests

gigitux committed:
update string

Aljullu reviewed (approved):
Thanks for the updates, @gigitux! I'm approving this PR as everything looks good, but as you mentioned, there are currently some tests which are failing. Hopefully they will pass after rebasing on top of the other PRs. :+1:

nielslange commented:
I've removed @woocommerce/rubik as a reviewer as this PR does not touch any checkout-related e2e tests.

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce into fix/migration-to-pw-pt-1

gigitux committed:
Merge branch 'fix/migration-to-pw-pt-1' of github.com:woocommerce/woocommerce into fix/migration-to-pw-pt2

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce into fix/migration-to-pw-pt2

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce into fix/migration-to-pw-pt2

gigitux committed:
fix E2E test

gigitux committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce into fix/migration-to-pw-pt2

gigitux commented:
With https://github.com/woocommerce/woocommerce/pull/46671/commits/4dc98ce6beff7d6f9d60b6de0c9cc4b4b8034fc9 I addressed some feedback that @WunderBart gave me on #46644

gigitux committed:
fix E2E tests
